pkg/services/ssosettings/strategies/oauth_strategy.go

```diff
@@ -51,13 +51,20 @@ func (s *OAuthStrategy) loadAllSettings() {
 	allProviders := append(ssosettings.AllOAuthProviders, social.GrafanaNetProviderName)
 	for _, provider := range allProviders {
 		settings := s.loadSettingsForProvider(provider)
-		if provider == social.GrafanaNetProviderName {
+		// This is required to support the legacy settings for the provider (auth.grafananet section)
+		// It will use the settings (and overwrite the current grafana_com settings) from auth.grafananet if
+		// the auth.grafananet section is enabled and the auth.grafana_com section is disabled.
+		if provider == social.GrafanaNetProviderName && s.shouldUseGrafanaNetSettings() && settings["enabled"] == true {
 			provider = social.GrafanaComProviderName
 		}
```

Comment:
I'm a bit confused about this condition. My interpretation: if the current provider is grafana_net and and grafana_net should be used and grafana_net settings are enabled then go with grafana_com. Shouldn't we go with grafana_net if all those conditions are met?

Reply:
`auth.grafananet` section is the legacy way (it is still used) to configure the GrafanaCom login on instances. The new way of configuring the GrafanaCom connector is to use the `auth.grafana_com` section, but in the code we only have the `GrafanaCom` connector and we need to be backward compatible and if only `auth.grafananet` is configured + enabled then the GrafanaCom provider should use the data from the `auth.grafananet` section.

To understand this you should check the previous implementation of this:
https://github.com/grafana/grafana/blob/ecb8447a7fbad847ad2df6831df9981d2774d95c/pkg/login/social/socialimpl/service.go#L76-L98

And please check the `TestSocialService_ProvideService_GrafanaComGrafanaNet` (https://github.com/grafana/grafana/pull/83479/files#diff-7756bd0bc367cec4bbf22aac1c341f45ada4d97faaed0f6d8d67dbed67f38370R94) tests that I added to verify the previous (when `ssoSettingsApi` is disabled) behaviour.